lib/action_controller/serialization.rb

```diff
@@ -56,10 +56,15 @@ def use_adapter?
 
     [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
       define_method renderer_method do |resource, options|
-        options.fetch(:serialization_context) do
-          options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, options)
+        options.freeze
+        serialization_options = options.deep_dup
```

Comment:
Somehow there was some issues with the `type` not being duped properly. Therefore the test failed cause the `type` method return the wrong value. I didn't dig to much into it though. I can try and spend some more time to figure out why it was duped properly.